task2/task2.js

```diff
@@ -0,0 +1,41 @@
+//a
+//window.x = 1;
+const bind =
+  (fn, context) =>
+  (...args) =>
+    fn.apply(context, [...args]);
```

Comment:
It's more than a tricky solution :)
Let's try to avoid using `apply`, `call` and `bind` functions inside.